Mathlib/Topology/Algebra/Order/Compact.lean

```diff
@@ -378,61 +382,89 @@ end ConditionallyCompleteLinearOrder
+    (ne_s : s.Nonempty) : IsLeast s (sInf s) :=
   ⟨hs.sInf_mem ne_s, (hs.isGLB_sInf ne_s).1⟩
 #align is_compact.is_least_Inf IsCompact.isLeast_sInf
 
-theorem IsCompact.isGreatest_sSup {s : Set α} (hs : IsCompact s) (ne_s : s.Nonempty) :
-    IsGreatest s (sSup s) :=
-  @IsCompact.isLeast_sInf αᵒᵈ _ _ _ _ hs ne_s
+theorem IsCompact.isGreatest_sSup [UpperClosedTopology α] {s : Set α} (hs : IsCompact s)
+    (ne_s : s.Nonempty) : IsGreatest s (sSup s) :=
+  IsCompact.isLeast_sInf (α := αᵒᵈ) hs ne_s
 #align is_compact.is_greatest_Sup IsCompact.isGreatest_sSup
 
-theorem IsCompact.exists_sInf_image_eq_and_le {s : Set β} (hs : IsCompact s) (ne_s : s.Nonempty)
-    {f : β → α} (hf : ContinuousOn f s) : ∃ x ∈ s, sInf (f '' s) = f x ∧ ∀ y ∈ s, f x ≤ f y :=
+theorem IsCompact.exists_sInf_image_eq_and_le [LowerClosedTopology α] {s : Set β}
+    (hs : IsCompact s) (ne_s : s.Nonempty) {f : β → α} (hf : ContinuousOn f s) :
+    ∃ x ∈ s, sInf (f '' s) = f x ∧ ∀ y ∈ s, f x ≤ f y :=
   let ⟨x, hxs, hx⟩ := (hs.image_of_continuousOn hf).sInf_mem (ne_s.image f)
   ⟨x, hxs, hx.symm, fun _y hy =>
     hx.trans_le <| csInf_le (hs.image_of_continuousOn hf).bddBelow <| mem_image_of_mem f hy⟩
 #align is_compact.exists_Inf_image_eq_and_le IsCompact.exists_sInf_image_eq_and_le
 
-theorem IsCompact.exists_sSup_image_eq_and_ge {s : Set β} (hs : IsCompact s) (ne_s : s.Nonempty)
-    {f : β → α} (hf : ContinuousOn f s) : ∃ x ∈ s, sSup (f '' s) = f x ∧ ∀ y ∈ s, f y ≤ f x :=
-  @IsCompact.exists_sInf_image_eq_and_le αᵒᵈ _ _ _ _ _ _ hs ne_s _ hf
+theorem IsCompact.exists_sSup_image_eq_and_ge [UpperClosedTopology α] {s : Set β}
+    (hs : IsCompact s) (ne_s : s.Nonempty) {f : β → α} (hf : ContinuousOn f s) :
+    ∃ x ∈ s, sSup (f '' s) = f x ∧ ∀ y ∈ s, f y ≤ f x :=
+  IsCompact.exists_sInf_image_eq_and_le (α := αᵒᵈ) hs ne_s hf
 #align is_compact.exists_Sup_image_eq_and_ge IsCompact.exists_sSup_image_eq_and_ge
 
-theorem IsCompact.exists_sInf_image_eq {s : Set β} (hs : IsCompact s) (ne_s : s.Nonempty)
-    {f : β → α} (hf : ContinuousOn f s) : ∃ x ∈ s, sInf (f '' s) = f x :=
+theorem IsCompact.exists_sInf_image_eq [LowerClosedTopology α] {s : Set β} (hs : IsCompact s)
+    (ne_s : s.Nonempty) {f : β → α} (hf : ContinuousOn f s) : ∃ x ∈ s, sInf (f '' s) = f x :=
   let ⟨x, hxs, hx, _⟩ := hs.exists_sInf_image_eq_and_le ne_s hf
   ⟨x, hxs, hx⟩
 #align is_compact.exists_Inf_image_eq IsCompact.exists_sInf_image_eq
 
-theorem IsCompact.exists_sSup_image_eq :
-    ∀ {s : Set β},
-      IsCompact s → s.Nonempty → ∀ {f : β → α}, ContinuousOn f s → ∃ x ∈ s, sSup (f '' s) = f x :=
-  @IsCompact.exists_sInf_image_eq αᵒᵈ _ _ _ _ _
+theorem IsCompact.exists_sSup_image_eq [UpperClosedTopology α] {s : Set β} (hs : IsCompact s)
+    (ne_s : s.Nonempty) : ∀ {f : β → α}, ContinuousOn f s → ∃ x ∈ s, sSup (f '' s) = f x :=
+  IsCompact.exists_sInf_image_eq (α := αᵒᵈ) hs ne_s
 #align is_compact.exists_Sup_image_eq IsCompact.exists_sSup_image_eq
 
+-- porting note: todo: add dual versions
+
```

Comment:
I know you didn't write this, but do you want to address it now while we're at it? If you want to put it in a dependent PR that's fine.

Reply:
Done, except for the deprecated one.